src/Sales/OrderBuilder.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+namespace TddWizard\Fixtures\Sales;
+
+use Magento\Sales\Api\Data\OrderInterface;
+use TddWizard\Fixtures\Catalog\ProductBuilder;
+use TddWizard\Fixtures\Checkout\CartBuilder;
+use TddWizard\Fixtures\Checkout\CustomerCheckout;
+use TddWizard\Fixtures\Customer\AddressBuilder;
+use TddWizard\Fixtures\Customer\CustomerBuilder;
+use TddWizard\Fixtures\Customer\CustomerFixture;
+
+/**
+ * Builder to be used by fixtures
+ */
+class OrderBuilder
+{
+    /**
+     * @var CartBuilder
+     */
+    private $cartBuilder;
+
+    /**
+     * @var ProductBuilder[]
+     */
+    private $productBuilders;
+
+    /**
+     * @var CustomerBuilder
+     */
+    private $customerBuilder;
+
+    /**
+     * @var string
+     */
+    private $shippingMethod;
+
+    /**
+     * @var string
+     */
+    private $paymentMethod;
+
+    public static function anOrder(): OrderBuilder
+    {
+        return new static();
+    }
+
+    public function withProducts(ProductBuilder ...$productBuilders): OrderBuilder
+    {
+        $builder = clone $this;
+        $builder->productBuilders = $productBuilders;
+
+        return $builder;
+    }
+
+    public function withCustomer(CustomerBuilder $customerBuilder): OrderBuilder
+    {
+        $builder = clone $this;
+        $builder->customerBuilder = $customerBuilder;
+
+        return $builder;
+    }
+
+    public function withCart(CartBuilder $cartBuilder): OrderBuilder
+    {
+        $builder = clone $this;
+        $builder->cartBuilder = $cartBuilder;
+
+        return $builder;
+    }
+
+    public function withShippingMethod(string $shippingMethod): OrderBuilder
+    {
+        $builder = clone $this;
+        $builder->shippingMethod = $shippingMethod;
+
+        return $builder;
+    }
+
+    public function withPaymentMethod(string $paymentMethod): OrderBuilder
+    {
+        $builder = clone $this;
+        $builder->paymentMethod = $paymentMethod;
+
+        return $builder;
+    }
+
+    /**
+     * @return OrderInterface
+     * @throws \Exception
+     */
+    public function build(): OrderInterface
+    {
+        $builder = clone $this;
+
+        if (empty($builder->productBuilders)) {
+            // init simple products
+            for ($i = 1; $i <= mt_rand(1, 5); $i++) {
```

Comment:
Can we agree on a fixed amount to keep tests as deterministic as possible? Otherwise there's a risk, they sometimes pass, sometimes fail depending on number of products.